src/pages/CourseContent/index.js

```diff
@@ -37,25 +37,57 @@ function CourseContent(props) {
   }, [dispatch, courseId]);
 
   useEffect(() => {
-    const firstExercise = get(data, "exerciseList[0]");
+    let defaultExercise, defaultExerciseIndex;
+    let exerciseIdFromParams = getExerciseIdFromUrl();
+    const firstExercise = get(data, 'exerciseList[0]');
+
+    // exercises loaded
     if (firstExercise) {
-      const selectedExerciseInfo = { exercise: firstExercise, index: 0 };
-      dispatch(courseActions.updateSelectedExercise(selectedExerciseInfo));
+
+        // set default exercise as per the exercise id present in the url
+        // TBD: Ideally, when navigating to a course content page, the first exercise should be pre-computed and the url should be course/:courseId/exercise/:exerciseId so we can always use the exerciseId from the params to set the default exercise.
+        if (exerciseIdFromParams) {
+            const exerciseFromParams = data.exerciseList.find((exercise) => {
+                return exercise.id === exerciseIdFromParams;
+            });
+            if (exerciseFromParams) {
+                defaultExercise = exerciseFromParams;
+                defaultExerciseIndex = data.exerciseList.findIndex((exercise) => {
```

Comment:
We might not need two loops to get the exercise from the `exerciseList` .  first, we get the index and then get it directly from the `exerciseList` by index access. i.e something like this. 
```
defaultExerciseIndex = data.exerciseList.findIndex((exercise) => {
                return exercise.id === exerciseIdFromParams;
            });
            defaultExercise = data.exerciseList[defaultExerciseIndex]
``` 
Let me know about your thoughts.

Reply:
agreed, not sure why i did the other way round 😅

Reply:
Cool. Man. I was the observer. It's easy for me to see that. I would have done the same I think. 😝